src/collectors/mesos/mesos.py

```diff
@@ -12,55 +12,187 @@
     host = localhost
     port = 5050
 ```
+#### Dependencies
+ * urlib2
 """
 
-import diamond.collector
-import json
 import urllib2
 
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+import diamond.collector
+
+from diamond.collector import str_to_bool
+
 
 class MesosCollector(diamond.collector.Collector):
+    def __init__(self, config=None, handlers=[], name=None, configfile=None):
+        self.master = True
+        self.known_frameworks = {}
+        super(MesosCollector, self).__init__(config, handlers, name, configfile)
 
-    METRICS_PATH = "metrics/snapshot"
+    def process_config(self):
+        super(MesosCollector, self).process_config()
+        self.master = str_to_bool(self.config['master'])
 
     def get_default_config_help(self):
         config_help = super(MesosCollector,
                             self).get_default_config_help()
         config_help.update({
             'host': 'Hostname',
-            'port': 'Port (default is 5050; please set to 5051 for mesos-slave)'
+            'port': 'Port (default is 5050; please set to 5051 for mesos-slave)',  # noqa
+            'master': 'True if host is master'
         })
         return config_help
 
     def get_default_config(self):
+        """
+        Returns the default collector settings
+        """
         config = super(MesosCollector, self).get_default_config()
         config.update({
-            'host': 'localhost',
+            'host': '127.0.0.1',
             'port': 5050,
-            'path': 'mesos'
+            'path': 'mesos',
+            'master': True
         })
         return config
 
-    def __init__(self, *args, **kwargs):
-        super(MesosCollector, self).__init__(*args, **kwargs)
-
     def collect(self):
-        metrics = self.get_metrics()
-
-        for k, v in metrics.iteritems():
-            key = self.clean_up(k)
-            self.publish(key, v)
-
-    def get_metrics(self):
+        if json is None:
+            self.log.error('Unable to import json')
+            return
+        self._collect_metrics_snapshot()
+        if not self.master:
+            self._collect_slave_state()
+            self._collect_slave_statistics()
+
+    def _collect_metrics_snapshot(self):
+        result = self._get(
+            self.config['host'],
+            self.config['port'],
+            'metrics/snapshot'
+        )
+        if not result:
+            return
+
+        for key in result:
+            value = result[key]
+            self.publish(key, value, precision=self._precision(value))
+
+    def _collect_slave_state(self):
+        result = self._get(
+            self.config['host'],
+            self.config['port'],
+            'slave(1)/state.json'
```

Comment:
Yes, it's working with 1.0 but this endpoint could be deleted. I assume nobody use this anciend version of Mesos and changed remove `.json` suffix.

[src/slave/slave.cpp:725-727](https://github.com/apache/mesos/blob/ca5642cf0986507884a3a94b6c500e2c48e81121/src/slave/slave.cpp#L725-L727)
[src/master/master.cpp:955-957](https://github.com/apache/mesos/blob/ca5642cf0986507884a3a94b6c500e2c48e81121/src/master/master.cpp#L955-L957)